### ISSUE TITLE
PyTrilinos: Syntax error while building

### ISSUE THREAD

**Issue by millist097** [trilinos/Trilinos#5999]
## Question
@trilinos/\<PyTrilinos\>

I am currently trying to install Trilinos on Arch Linux. It builds fine until it reaches PyTrilinos. It then fails with the following error

```
[ 94%] Building CXX object packages/PyTrilinos/src/CMakeFiles/pytrilinos.dir/PyTrilinos_Tpetra_Util.cpp.o
[ 94%] Building CXX object packages/PyTrilinos/src/CMakeFiles/pytrilinos.dir/PyTrilinos_Domi_Util.cpp.o
/home/glitch/Programs/Trilinos/packages/tpetra/core/src/Tpetra_Map_decl.hpp:1545: Error: Syntax error - possibly a missing semicolon.
make[2]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Teuchos_RCP.dir/build.make:62: packages/PyTrilinos/src/Teuchos.RCPPYTHON_wrap.cpp] Error 1
make[1]: *** [CMakeFiles/Makefile2:12689: packages/PyTrilinos/src/CMakeFiles/PyTrilinos_Teuchos_RCP.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

```
The options I have are as follows:

cmake \
-D CMAKE_BUILD_TYPE:STRING=RELEASE \
-D Trilinos_ENABLE_PyTrilinos:BOOL=ON \
-D BUILD_SHARED_LIBS:BOOL=ON \
-D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON \
-D DTrilinos_ENABLE_TESTS:BOOL=ON \
-D DART_TESTING_TESTING_TIMEOUT:STRING=600 \
-D DMPI_BASE_DIR=/home/glitch/.openmpi 

I have been looking through issues and post but have not found anything yet. I am not sure if this is something with my system or if I am using something I shouldn't be.

Environment:
Operating System: Arch Linux
   Kernel: Linux 4.19.73-1-lts
   Architecture: x86-64
Python version is 3.7.4
CPU: AMD Ryzen 5 1400
gcc version 9.1.0 
cmake version 3.15.3

I am still a bit new at this so I am not sure if this is the correct place for this question.
Thank you in advance for any help. If I have done something wrong please let me know.

**Comment by mhoemmen** [comment]
What version of Trilinos?  If you don't need PyTrilinos, you can disable the package for now: `Trilinos_ENABLE_PyTrilinos:BOOL=OFF`.

**Comment by millist097** [comment]
The Version is 12.12

if I could I would but it happens that PyTrilinos is the package I need.

**Comment by mhoemmen** [comment]
@millist097 That version is several years old.  Have you tried Trilinos' master branch, or even `trilinos-release-12-16-branch`?

**Comment by jhux2** [comment]
@trilinos/trilinos-product-owners

**Comment by millist097** [comment]
Ok so I was wrong when I said I was using version 12.12 that was the version listed in the release notes. I have been pulling  the master branch. (the master branch generates the error above)

I did try switching to the 12-16 branch but swapped one error for another

```
[ 90%] Building CXX object packages/PyTrilinos/src/CMakeFiles/pytrilinos.dir/PyTrilinos_EpetraExt_Util.cpp.o
[ 90%] Building CXX object packages/PyTrilinos/src/CMakeFiles/pytrilinos.dir/PyTrilinos_ML_Util.cpp.o
[ 90%] Linking CXX shared library libpytrilinos.so
[ 90%] Built target pytrilinos
[ 90%] Swig source /home/glitch/Programs/Trilinos/packages/PyTrilinos/src/LOCA.Epetra.Interface.i
Deprecated command line option: -noproxydel. This option is now always on.
/home/glitch/Programs/Trilinos/packages/PyTrilinos/src/Teuchos.i:148: Error: Syntax error in input(1).
make[2]: *** [packages/PyTrilinos/src/CMakeFiles/PyTrilinos_LOCA_Epetra_Interface.dir/build.make:62: packages/PyTrilinos/src/LOCA.Epetra.InterfacePYTHON_wrap.cpp] Error 1
make[1]: *** [CMakeFiles/Makefile2:10619: packages/PyTrilinos/src/CMakeFiles/PyTrilinos_LOCA_Epetra_Interface.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

**Comment by pyt-viper** [comment]
Hopefully, these issues are addressed by PR #6501

**Comment by pyt-viper** [comment]
@millist097 can you check to see if the latest develop branch addresses your problem?
